crypto/evp/digest.c

```diff
@@ -335,11 +335,18 @@ int EVP_DigestFinal(EVP_MD_CTX *ctx, unsigned char *md, unsigned int *size)
 /* The caller can assume that this removes any secret data from the context */
 int EVP_DigestFinal_ex(EVP_MD_CTX *ctx, unsigned char *md, unsigned int *isize)
 {
-    int ret;
+    int ret, sz;
     size_t size = 0;
-    size_t mdsize = EVP_MD_size(ctx->digest);
+    size_t mdsize = 0;
 
-    if (ctx->digest == NULL || ctx->digest->prov == NULL)
+    if (ctx->digest == NULL)
+        return 0;
+
+    sz = EVP_MD_size(ctx->digest);
+    if (sz < 0)
+        return 0;
+    mdsize = sz;
```

Comment:
I see no reason for the `sz` variable, just assign it to `mdsize`.

EVP_MD_size() is not documented as being able to return an error.

Reply:
Oh, EVP_MD_size() of course returns an int, but mdsize is a size_t.

Reply:
yep.. There are a few of these.